src/Aquifer.API/Endpoints/Resources/ResourceReferences/Create/Endpoint.cs

```diff
@@ -0,0 +1,45 @@
+using Aquifer.API.Common;
+using Aquifer.Data;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.API.Endpoints.Resources.ResourceReferences.Create;
+
+public class Endpoint(AquiferDbContext dbContext) : Endpoint<Request>
+{
+    public override void Configure()
+    {
+        Post("/resources/resource-references");
+        Permissions(PermissionName.EditContentResourceReferences);
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
+    {
+        var resourceContent = await dbContext.ResourceContents
+            .Include(rc => rc.Resource)
+            .ThenInclude(r => r.AssociatedResourceChildren)
+            .SingleOrDefaultAsync(rc => rc.Id == request.ResourceContentId, ct);
+
+        if (resourceContent == null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        var referenceResource = await dbContext.Resources.FindAsync([request.ReferenceResourceId, ct], ct);
+
+        if (referenceResource == null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        if (!resourceContent.Resource.AssociatedResourceChildren.Any(r => r.Id == request.ReferenceResourceId))
```

Comment:
I'd wrap this conditional around everything with `referenceResource`. If this is false, then there's no reason to go to the database to grab `referenceResource` in the first place.

Reply:
Also seems like the `SaveChangesAsync` would be in the conditional also, since no reason to save if nothing's added.